recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,155 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class MongoCDriverConan(ConanFile):
+    name = "mongo-c-driver"
+    license = "Apache License 2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://mongoc.org/"
+    description = "A Cross Platform MongoDB Client Library for C"
+    topics = ("conan", "libbson", "libmongoc", "mongo", "mongodb", "database", "db")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "ssl_provider": ['DARWIN', 'WINDOWS', 'OPENSSL', 'LIBRESSL', 'OFF'],
+        "zlib": ['SYSTEM', 'BUNDLED', 'OFF']
```

Comment:
Similar to above, this is typically `with_zlib : true, false` and we always offer up the Conan version

Can you please explain, https://github.com/conan-io/conan-center-index/pull/3451/files#diff-4c1fc5f45362325231b6006dca7417f7135b73a7ae89c1691954c51648a89233R110

> Usage zlib provided by conan on Windows can be broken

😟

Reply:
```zlib.h``` can't be found. [See build log](https://storage.googleapis.com/gitlab-gprd-artifacts/4f/8e/4f8e4caf7bb305fa32155dffe338c76a138a9775422a186b1c1e6caae4dd3a1a/2020_11_07/836269468/918675914/job.log?response-content-type=text%2Fplain%3B%20charset%3Dutf-8&response-content-disposition=inline&GoogleAccessId=gitlab-object-storage-prd@gitlab-production.iam.gserviceaccount.com&Signature=pWnzBCuJTMhlOm8zLS849jfjeCMc9bs%2BhdoM68cLfSXiCT5twWGoJxVSeLGf%0Af4ZSxWfVQPalFmPo6q2GKKXSWGWWqA3%2FaUvIf0MDaSNiklPnX4gegVnvOdSp%0AWdhMCamh5GIdZTFoyTp%2BeZIiUlcOVXg76WEsYIf5Yt3AKPFxZPNStaBrkg4a%0Afk1E7BNsihLTZkdu%2FmyjzGRMxKyFOTbSS99rMtvAbu7S0y5X79tMjlqPHR6E%0ADYa%2BKiuLf4Vr9dGTzedrC7Csyn0KHcH97UYdtZyLAjaE6Ex5lO%2FL3RC%2B1DG0%0AtEueZQZoaCmfVYk8p9BSIaJRBZt0DDbRynEY6gzziA%3D%3D&Expires=1604846173)

Reply:
>  we always offer up the Conan version

Yes, i agree, but it's not working...

Reply:
You may have to patch upstream CMakeLists, and add `cmake_find_package` generator in this recipe.

Reply:
addressed in n-bes#1